pkg/primitives/vm.go

```diff
@@ -280,6 +282,29 @@ func (p *Primitives) virtualMachineProvisionImpl(ctx context.Context, wl *gridty
 		if err := p.vmMounts(ctx, deployment, config.Mounts, true, &machine); err != nil {
 			return result, err
 		}
+		if config.Corex {
+			rootBinPath := filepath.Join(mnt, "bin")
+			_, err = os.Stat(rootBinPath)
+			if err != nil {
```

Comment:
I don't get this check `mnt` exists for sure (granted by the success of the flist mount). the `<rootfs>/bin` on other hand is not granted to exist because it depends on the flist (what if the flist doesn't have bin) this should NOT stop the corex from working. If we want to avoid crating this bin folder under the container <rootfs>. We can simply just copy the corex binary to <rootfs>/ directly. Hence your entrypoint is going to be `/corex` not `/bin/corex`. Which I think is the right way.